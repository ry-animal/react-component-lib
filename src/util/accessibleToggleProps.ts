export function accessibleToggleProps(
  label: string,
  controls: string,
  expanded: boolean,
  popupType?: "menu"
) {
  return {
    "aria-controls": controls,
    "aria-expanded": expanded,
    "aria-haspopup": popupType || true,
    "aria-label": label,
    "aria-pressed": expanded,
    role: "button",
  };
}
