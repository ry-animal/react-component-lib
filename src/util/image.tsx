import { ISelectedFile } from "src/components/Forms/FileUpload/ISelectedFile";

export const makeDataUrl = (selectedFile: ISelectedFile) =>
  URL.createObjectURL(selectedFile.file);
