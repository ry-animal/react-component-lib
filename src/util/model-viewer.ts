/*
 * This file is a hacky way of allowing any other component to
 * use the <model-viewer /> tag just by doing:
 *   import "../../../util/model-viewer";
 *
 * The authors of the package do not expose it in a React-friendly
 * way, see: https://github.com/google/model-viewer/issues/1502
 *
 * The component documentation is at https://modelviewer.dev/docs/
 */

import "@google/model-viewer";

declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace JSX {
    export interface IntrinsicElements {
      "model-viewer": {
        src?: string;
        alt?: string;
        poster?: string;
        autoplay?: boolean;
        exposure?: number;
        "camera-controls"?: boolean;
        "environment-image"?: string;
        "seamless-poster"?: string;
        loading?: "auto" | "lazy" | "eager";
        reveal?: "auto" | "interaction" | "manual";
      } & React.DetailedHTMLProps<
        React.HTMLAttributes<HTMLElement>,
        HTMLElement
      >;
    }
  }
}
