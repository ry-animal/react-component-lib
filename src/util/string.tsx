const DEFAULT_LOCALE = "en-US";

export const truncateEthAddress = (str: string): string => {
  if (str.length) {
    const numStartingCharacters = 6;
    const numTrailingCharacters = 4;
    return `${str.substring(0, numStartingCharacters)}...${str.substring(
      str.length - numTrailingCharacters,
      str.length
    )}`;
  }
  return "";
};

export const formatUSDPriceString = (value: number): string => {
  const minimumFractionDigits = 2;
  const maximumFractionDigits = 2;
  return `$${value.toLocaleString(DEFAULT_LOCALE, {
    minimumFractionDigits,
    maximumFractionDigits,
  })}`;
};

export const formatETHPriceString = (priceInEth: number): string => {
  const minimumFractionDigits = 0;
  const maximumFractionDigits = 8;
  const maxDigitsWithDecimal = 8;

  let formatted = `${priceInEth
    .toLocaleString(DEFAULT_LOCALE, {
      minimumFractionDigits,
      maximumFractionDigits,
    })
    .replace(/,/gu, "")
    .substring(0, maxDigitsWithDecimal)}`;

  if (formatted.endsWith(".")) {
    formatted = formatted.slice(0, -1);
  }
  return formatted;
};

const evaluateNumberLessThanOne = (num: number) => {
  if (num === 0) {
    return "0";
  }

  const minDecimalValue = 0.0001;
  if (num < minDecimalValue) {
    return minDecimalValue.toString();
  }
  // To round decimal to 4 digits
  const maxDigits = 10000;
  return `${Math.round(num * maxDigits) / maxDigits}`;
};

// Abbreviate Number https://stackoverflow.com/a/9462382

export const abbreviateNumber = (num: number): string => {
  const lookup = [
    { value: 1, symbol: "" },
    { value: 1e3, symbol: "k" },
    { value: 1e6, symbol: "M" },
    { value: 1e9, symbol: "B" },
    { value: 1e12, symbol: "T" },
    { value: 1e15, symbol: "P" },
    { value: 1e18, symbol: "E" },
  ];

  const rx = /\.0+$|(?<id>\.[0-9]*[1-9])0+$/u;
  const digits = 1;

  const item = lookup
    .slice()
    .reverse()
    .find((x) => num >= x.value);
  return item
    ? (num / item.value).toFixed(digits).replace(rx, "$1") + item.symbol
    : evaluateNumberLessThanOne(num);
};

export const getDomainFromRegex = (url: string) => {
  const matches = url.match(/:\/\/(?<id>.[^/]+)/u);
  // Extract domain (will be null if no match is found)
  return matches && matches[1];
};
