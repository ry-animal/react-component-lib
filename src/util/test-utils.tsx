import { render, RenderOptions } from "@testing-library/react";
import "jest-styled-components";
import * as React from "react";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { LIGHT_THEME } from "../theme";

interface ProvidersProps {
  children: React.ReactNode;
}

function Providers({ children }: ProvidersProps) {
  return (
    <BrowserRouter>
      <ThemeProvider theme={LIGHT_THEME}>{children}</ThemeProvider>
    </BrowserRouter>
  );
}

const customRender = (
  ui: JSX.Element,
  options?:
    | RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement
      >
    | undefined
) => render(ui, { wrapper: Providers, ...options });

// Duplicated instead of import "../../__mocks__/matchMediaMock"; because jest has trouble resolving imports
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

// Re-export everything
export * from "@testing-library/react";
// Override render method
export { customRender as render };
