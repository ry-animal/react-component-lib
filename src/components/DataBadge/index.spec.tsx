import { DataBadge } from ".";
import { render } from "../../util/test-utils";

describe("<DataBadge />", () => {
  const baseProps = {
    title: "Foo",
    name: "Bar",
    imageSrc: "//source.unsplash.com/user/erondu/200x200",
  };

  it("should render", () => {
    expect(render(<DataBadge {...baseProps} />).asFragment()).toMatchSnapshot();
  });
  it("renders without name", () => {
    expect(
      render(<DataBadge imageSrc={baseProps.imageSrc} />).asFragment()
    ).toMatchSnapshot();
  });
  it("renders size m", () => {
    expect(
      render(<DataBadge {...baseProps} size="m" />).asFragment()
    ).toMatchSnapshot();
  });
  it("renders size l", () => {
    expect(
      render(<DataBadge {...baseProps} size="l" />).asFragment()
    ).toMatchSnapshot();
  });
  it("renders with external link", () => {
    expect(
      render(<DataBadge {...baseProps} hasLinkIcon />).asFragment()
    ).toMatchSnapshot();
  });
  it("renders with border", () => {
    expect(
      render(<DataBadge {...baseProps} hasBorder />).asFragment()
    ).toMatchSnapshot();
  });
  it("renders inline", () => {
    expect(
      render(<DataBadge {...baseProps} inline />).asFragment()
    ).toMatchSnapshot();
  });
  it("renders variant", () => {
    expect(
      render(
        <DataBadge {...baseProps} isDark imageSrc={baseProps.imageSrc} />
      ).asFragment()
    ).toMatchSnapshot();
  });
  it("renders hexagon", () => {
    expect(
      render(
        <DataBadge name="foo" isImageHexagon showImageFallback />
      ).asFragment()
    ).toMatchSnapshot();
  });
});
