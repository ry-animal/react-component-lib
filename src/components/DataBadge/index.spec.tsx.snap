// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataBadge /> renders hexagon 1`] = `
<DocumentFragment>
  .c3 {
  position: relative;
  height: 40px;
  width: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-clip-path: url(#hexPath);
  clip-path: url(#hexPath);
}

.c3:before {
  content: "";
  background-color: white;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  position: absolute;
}

.c4 {
  background-color: #EEEEEE;
  background-size: cover;
  background-position: center;
  position: relative;
  height: 40px;
  width: 40px;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2 {
  position: absolute;
}

.c5 {
  -webkit-clip-path: url(#hexPath);
  clip-path: url(#hexPath);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 100%;
}

.c1 {
  overflow: hidden;
  margin-right: 8px;
}

.c1 img {
  max-width: 100%;
}

.c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 16px;
  overflow: hidden;
  margin: 0;
  font-weight: 700;
  color: #151515;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 12px;
}

.c7 {
  display: block;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c8 {
  font-weight: 400;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <svg
          fill="none"
          height="0"
          viewBox="0 0 50 50"
          width="0"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <clippath
              clipPathUnits="objectBoundingBox"
              id="hexPath"
              transform="scale(0.02, 0.02)"
            >
              <path
                d="M26.2,0.7l19.2,11.1c0.7,0.4,1.2,1.2,1.2,2.1v22.2c0,0.9-0.5,1.7-1.2,2.1L26.2,49.3c-0.7,0.4-1.7,0.4-2.4,0L4.5,38.2c-0.7-0.4-1.2-1.2-1.2-2.1V13.9c0-0.9,0.5-1.7,1.2-2.1L23.8,0.7C24.5,0.3,25.5,0.3,26.2,0.7z"
              />
            </clippath>
          </defs>
        </svg>
      </div>
      <div
        class="c3"
        size="40"
      >
        <div
          class="c4 c5"
          size="40"
        />
      </div>
    </div>
    <p
      class="c6"
    >
      <span
        class="c7 c8"
      >
        foo
      </span>
    </p>
  </div>
</DocumentFragment>
`;

exports[`<DataBadge /> renders inline 1`] = `
<DocumentFragment>
  .c3 {
  background-image: url("//source.unsplash.com/user/erondu/200x200");
  background-color: #EEEEEE;
  background-size: cover;
  background-position: center;
  position: relative;
  height: 40px;
  width: 40px;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c4 {
  border-radius: 50%;
}

.c2 {
  position: relative;
  height: 40px;
  width: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-radius: 50%;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2:before {
  content: "";
  background-color: white;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  position: absolute;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 100%;
  margin: 4px 8px 4px 0;
}

.c1 {
  overflow: hidden;
  margin-right: 8px;
}

.c1 img {
  max-width: 100%;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 16px;
  overflow: hidden;
  margin: 0;
  font-weight: 700;
  color: #151515;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  font-size: 12px;
}

.c6 {
  display: block;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c7 {
  font-weight: 400;
  padding-left: 12px;
  font-weight: 500;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        size="40"
      >
        <div
          class="c3 c4"
          size="40"
        />
      </div>
    </div>
    <p
      class="c5"
    >
      <span
        class="c6"
      >
        Foo
      </span>
      <span
        class="c6 c7"
      >
        Bar
      </span>
    </p>
  </div>
</DocumentFragment>
`;

exports[`<DataBadge /> renders size l 1`] = `
<DocumentFragment>
  .c3 {
  background-image: url("//source.unsplash.com/user/erondu/200x200");
  background-color: #EEEEEE;
  background-size: cover;
  background-position: center;
  position: relative;
  height: 80px;
  width: 80px;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c4 {
  border-radius: 50%;
}

.c2 {
  position: relative;
  height: 80px;
  width: 80px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-radius: 50%;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2:before {
  content: "";
  background-color: white;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  position: absolute;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 100%;
}

.c1 {
  overflow: hidden;
  margin-right: 8px;
}

.c1 img {
  max-width: 100%;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 16px;
  overflow: hidden;
  margin: 0;
  font-weight: 700;
  color: #151515;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 12px;
}

.c6 {
  display: block;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c7 {
  font-weight: 400;
  font-weight: 500;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        size="80"
      >
        <div
          class="c3 c4"
          size="80"
        />
      </div>
    </div>
    <p
      class="c5"
    >
      <span
        class="c6"
      >
        Foo
      </span>
      <span
        class="c6 c7"
      >
        Bar
      </span>
    </p>
  </div>
</DocumentFragment>
`;

exports[`<DataBadge /> renders size m 1`] = `
<DocumentFragment>
  .c3 {
  background-image: url("//source.unsplash.com/user/erondu/200x200");
  background-color: #EEEEEE;
  background-size: cover;
  background-position: center;
  position: relative;
  height: 60px;
  width: 60px;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c4 {
  border-radius: 50%;
}

.c2 {
  position: relative;
  height: 60px;
  width: 60px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-radius: 50%;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2:before {
  content: "";
  background-color: white;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  position: absolute;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 100%;
}

.c1 {
  overflow: hidden;
  margin-right: 8px;
}

.c1 img {
  max-width: 100%;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 16px;
  overflow: hidden;
  margin: 0;
  font-weight: 700;
  color: #151515;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 12px;
}

.c6 {
  display: block;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c7 {
  font-weight: 400;
  font-weight: 500;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        size="60"
      >
        <div
          class="c3 c4"
          size="60"
        />
      </div>
    </div>
    <p
      class="c5"
    >
      <span
        class="c6"
      >
        Foo
      </span>
      <span
        class="c6 c7"
      >
        Bar
      </span>
    </p>
  </div>
</DocumentFragment>
`;

exports[`<DataBadge /> renders variant 1`] = `
<DocumentFragment>
  .c3 {
  background-image: url("//source.unsplash.com/user/erondu/200x200");
  background-color: #EEEEEE;
  background-size: cover;
  background-position: center;
  position: relative;
  height: 40px;
  width: 40px;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c4 {
  border-radius: 50%;
}

.c2 {
  position: relative;
  height: 40px;
  width: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-radius: 50%;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2:before {
  content: "";
  background-color: white;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  position: absolute;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 100%;
}

.c1 {
  overflow: hidden;
  margin-right: 8px;
}

.c1 img {
  max-width: 100%;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 16px;
  overflow: hidden;
  margin: 0;
  font-weight: 400;
  color: #ffffff;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 12px;
}

.c6 {
  display: block;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c7 {
  font-weight: 400;
  font-weight: 500;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        size="40"
      >
        <div
          class="c3 c4"
          size="40"
        />
      </div>
    </div>
    <p
      class="c5"
    >
      <span
        class="c6"
      >
        Foo
      </span>
      <span
        class="c6 c7"
      >
        Bar
      </span>
    </p>
  </div>
</DocumentFragment>
`;

exports[`<DataBadge /> renders with border 1`] = `
<DocumentFragment>
  .c3 {
  background-image: url("//source.unsplash.com/user/erondu/200x200");
  background-color: #EEEEEE;
  background-size: cover;
  background-position: center;
  position: relative;
  height: 40px;
  width: 40px;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c4 {
  border-radius: 50%;
}

.c2 {
  position: relative;
  height: 40px;
  width: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-radius: 50%;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2:before {
  content: "";
  background-color: white;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  position: absolute;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 100%;
  border: 1px solid #DBDBDB;
  background-color: #FFFFFF;
  padding: 8px;
  border-radius: 4px;
}

.c1 {
  overflow: hidden;
  margin-right: 8px;
}

.c1 img {
  max-width: 100%;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 16px;
  overflow: hidden;
  margin: 0;
  font-weight: 700;
  color: #151515;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 12px;
}

.c6 {
  display: block;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c7 {
  font-weight: 400;
  font-weight: 500;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        size="40"
      >
        <div
          class="c3 c4"
          size="40"
        />
      </div>
    </div>
    <p
      class="c5"
    >
      <span
        class="c6"
      >
        Foo
      </span>
      <span
        class="c6 c7"
      >
        Bar
      </span>
    </p>
  </div>
</DocumentFragment>
`;

exports[`<DataBadge /> renders with external link 1`] = `
<DocumentFragment>
  .c3 {
  background-image: url("//source.unsplash.com/user/erondu/200x200");
  background-color: #EEEEEE;
  background-size: cover;
  background-position: center;
  position: relative;
  height: 40px;
  width: 40px;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c4 {
  border-radius: 50%;
}

.c2 {
  position: relative;
  height: 40px;
  width: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-radius: 50%;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2:before {
  content: "";
  background-color: white;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  position: absolute;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 100%;
}

.c1 {
  overflow: hidden;
  margin-right: 8px;
}

.c1 img {
  max-width: 100%;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 16px;
  overflow: hidden;
  margin: 0;
  font-weight: 700;
  color: #151515;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 12px;
}

.c6 {
  display: block;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c7 {
  font-weight: 400;
  padding-right: 12px;
  font-weight: 500;
}

.c8 {
  padding-top: 16px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        size="40"
      >
        <div
          class="c3 c4"
          size="40"
        />
      </div>
    </div>
    <p
      class="c5"
    >
      <span
        class="c6"
      >
        Foo
      </span>
      <span
        class="c6 c7"
      >
        Bar
      </span>
    </p>
    <span
      class="c8"
    >
      <svg
        fill="none"
        height="14"
        viewBox="0 0 14 13"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.911 11.5556H1.79991V1.44444H6.85547V0H1.79991C0.998247 0 0.355469 0.65 0.355469 1.44444V11.5556C0.355469 12.35 0.998247 13 1.79991 13H11.911C12.7055 13 13.3555 12.35 13.3555 11.5556V6.5H11.911V11.5556ZM8.29991 0V1.44444H10.8927L3.79325 8.54389L4.81158 9.56222L11.911 2.46278V5.05556H13.3555V0H8.29991Z"
          fill="#151515"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;

exports[`<DataBadge /> renders without name 1`] = `
<DocumentFragment>
  .c3 {
  background-image: url("//source.unsplash.com/user/erondu/200x200");
  background-color: #EEEEEE;
  background-size: cover;
  background-position: center;
  position: relative;
  height: 40px;
  width: 40px;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c4 {
  border-radius: 50%;
}

.c2 {
  position: relative;
  height: 40px;
  width: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-radius: 50%;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2:before {
  content: "";
  background-color: white;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  position: absolute;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 100%;
}

.c1 {
  overflow: hidden;
  margin-right: 8px;
}

.c1 img {
  max-width: 100%;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 16px;
  overflow: hidden;
  margin: 0;
  font-weight: 700;
  color: #151515;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 12px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        size="40"
      >
        <div
          class="c3 c4"
          size="40"
        />
      </div>
    </div>
    <p
      class="c5"
    />
  </div>
</DocumentFragment>
`;

exports[`<DataBadge /> should render 1`] = `
<DocumentFragment>
  .c3 {
  background-image: url("//source.unsplash.com/user/erondu/200x200");
  background-color: #EEEEEE;
  background-size: cover;
  background-position: center;
  position: relative;
  height: 40px;
  width: 40px;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c4 {
  border-radius: 50%;
}

.c2 {
  position: relative;
  height: 40px;
  width: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-radius: 50%;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2:before {
  content: "";
  background-color: white;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  position: absolute;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 100%;
}

.c1 {
  overflow: hidden;
  margin-right: 8px;
}

.c1 img {
  max-width: 100%;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 16px;
  overflow: hidden;
  margin: 0;
  font-weight: 700;
  color: #151515;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 12px;
}

.c6 {
  display: block;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c7 {
  font-weight: 400;
  font-weight: 500;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        size="40"
      >
        <div
          class="c3 c4"
          size="40"
        />
      </div>
    </div>
    <p
      class="c5"
    >
      <span
        class="c6"
      >
        Foo
      </span>
      <span
        class="c6 c7"
      >
        Bar
      </span>
    </p>
  </div>
</DocumentFragment>
`;
